### PR TITLE
Add indices metrics for each index mode

### PR DESCRIPTION
**Pull request by dnhatn**
This change introduces index metrics per node, grouped by by index mode. For each index mode, we track the number of indices, document count, and store size. These metrics will help compare the usage of logsdb and time_series indices to standard indices.

Other metrics, such as index longevity and newly created indices, could be added in a follow-up.

Here is the list of 9 metrics introduced in this PR:
- es.indices.standard.total
- es.indices.standard.docs.total
- es.indices.standard.bytes.total

- es.indices.time_series.total
- es.indices.time_series.docs.total
- es.indices.time_series.bytes.total

- es.indices.logsdb.total
- es.indices.logsdb.docs.total
- es.indices.logsdb.bytes.total

### CODE REVIEW

File: server/src/main/java/org/elasticsearch/monitor/metrics/IndicesMetrics.java [hunk 1]
```diff
@@ -0,0 +1,169 @@
...
+                        continue; // skip system indices
+                    }
+                    final IndexMode indexMode = indexShard.indexSettings().getMode();
+                    final IndexStats indexStats = stats.get(indexMode);
+                    final ShardRouting shardRouting = indexShard.routingEntry();
+                    if (shardRouting.primary() && shardRouting.shardId().id() == 0) {
+                        indexStats.numIndices++;
+                    }
+                    try {
+                        indexStats.numDocs += indexShard.commitStats().getNumDocs();
```
Comment (martijnvg) [inline server/src/main/java/org/elasticsearch/monitor/metrics/IndicesMetrics.java:145]: should we only keep track of these stats for primary shards?

File: server/src/main/java/org/elasticsearch/monitor/metrics/IndicesMetrics.java [hunk 2]
```diff
@@ -0,0 +1,169 @@
...
+ */
+public class IndicesMetrics extends AbstractLifecycleComponent {
+    private final Logger logger = LogManager.getLogger(IndicesMetrics.class);
+    private final MeterRegistry registry;
+    private final List<AutoCloseable> metrics = new ArrayList<>();
+    private final IndicesStatsCache stateCache;
+
+    public IndicesMetrics(MeterRegistry meterRegistry, IndicesService indicesService, TimeValue metricsInterval) {
+        this.registry = meterRegistry;
+        var cacheExpiry = new TimeValue(metricsInterval.getMillis() / 2);
```
Comment (martijnvg) [inline server/src/main/java/org/elasticsearch/monitor/metrics/IndicesMetrics.java:46]: Why `metricsInterval.getMillis() / 2`? Is the intent that each time metric interval has fresh metrics?

**Review by martijnvg — COMMENTED** [review]
This looks good Nhat! I left two questions.

File: server/src/main/java/org/elasticsearch/monitor/metrics/IndicesMetrics.java (hunk 2 above)
Comment (dnhatn) [reply server/src/main/java/org/elasticsearch/monitor/metrics/IndicesMetrics.java:48]: I've added a comment in https://github.com/elastic/elasticsearch/pull/113737/commits/9523b6e8ee5b13dfe4e894e999d3a919e2a52969.

File: server/src/main/java/org/elasticsearch/monitor/metrics/IndicesMetrics.java (hunk 1 above)
Comment (dnhatn) [reply server/src/main/java/org/elasticsearch/monitor/metrics/IndicesMetrics.java:147]: This works for me as well. If we decide to use only primaries, I think we should exclude relocating shards.

Comment (dnhatn) [reply server/src/main/java/org/elasticsearch/monitor/metrics/IndicesMetrics.java:153]: I pushed https://github.com/elastic/elasticsearch/pull/113737/commits/44456fa0d024eb641a230a45b2a388e77f269f41

Comment (martijnvg) [reply server/src/main/java/org/elasticsearch/monitor/metrics/IndicesMetrics.java:153]: 👍

**Review by martijnvg — APPROVED** [review]
LGTM

**Comment by dnhatn** [comment]
Thanks Martijn!

**Pull request merged**